src/nv_ingest/api/v2/ingest.py

```diff
@@ -0,0 +1,507 @@
+                        if isinstance(fetch_err, TimeoutError):
+                            raise HTTPException(status_code=410, detail="Job result is gone (TTL expired).")
+                        else:
+                            raise HTTPException(
+                                status_code=503, detail="Backend service unavailable preventing access to job result."
+                            )
+                    elif current_state == STATE_RETRIEVED_CACHED:
+                        raise HTTPException(
+                            status_code=410, detail="Job result is gone (previously cached, fetch failed)."
+                        )
+                    elif current_state in INTERMEDIATE_STATES:
+                        if isinstance(fetch_err, TimeoutError):
+                            raise HTTPException(
+                                status_code=202, detail=f"Job is processing (state: {current_state}). Retry later."
+                            )
+                        else:
+                            raise HTTPException(
+                                status_code=503, detail="Backend service unavailable preventing fetch of job result."
+                            )
+                    else:
+                        logger.error(f"Unexpected state '{current_state}' for job {job_id} after fetch failure.")
+                        raise HTTPException(
+                            status_code=500, detail="Internal server error: Unexpected job state after fetch failure."
+                        )
+            else:
+                logger.error(f"Unknown job state '{current_state}' for job {job_id}.")
+                raise HTTPException(
+                    status_code=500, detail=f"Internal server error: Unknown job state '{current_state}'."
+                )
+
+        else:
+            # This is a parent job - need to aggregate subjobs
+            subjob_ids = subjob_info.get("subjob_ids", [])
+            metadata = subjob_info.get("metadata", {})
+
+            logger.info(f"Parent job {job_id} has {len(subjob_ids)} subjobs")
+
+            total_pages = metadata.get("total_pages")
+            if isinstance(total_pages, str):
+                try:
+                    total_pages = int(total_pages)
+                except ValueError:
+                    logger.warning("Invalid total_pages '%s' for parent %s; recomputing", total_pages, job_id)
+                    total_pages = None
+
+            ordered_descriptors: List[Dict[str, Any]] = []
+            for idx, subjob_id in enumerate(subjob_ids, 1):
+                ordered_descriptors.append({"job_id": subjob_id, "chunk_index": idx})
+
+            # Check all subjob states
+            all_complete = True
+            any_failed = False
+            subjob_results = []
+            failed_subjobs: List[Dict[str, object]] = []
+
+            for page_index, descriptor in enumerate(ordered_descriptors, 1):
```

Comment:
This can be made parallel, I would issue all the subjob status requests together then process them as the results become available. `asyncio.gather`